From c6734719f3998aa5aa3ff3a42dfe41d7ad6a21b0 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Wed, 29 Oct 2014 16:33:53 -0400
Subject: [fs] ext4: remove obsoleted check

Message-id: <1414600485-1929-4-git-send-email-lczerner@redhat.com>
Patchwork-id: 98542
O-Subject: [RHEL 7.1 PATCH v2 03/55] ext4: remove obsoleted check
Bugzilla: 1156096
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Dmitry Monakhov <dmonakhov@openvz.org>

BZ 1156096
https://bugzilla.redhat.com/show_bug.cgi?id=1156096

BREW 8169722
https://brewweb.devel.redhat.com/taskinfo?taskID=8169722

Upstream commit 236f5ecb4a5876977837376729a7599cc1ab2cc3

 BH can not be NULL at this point, ext4_read_dirblock() always return
 non null value, and we already have done all necessery checks.

 Signed-off-by: Dmitry Monakhov <dmonakhov@openvz.org>
 Signed-off-by: Theodore Ts'o <tytso@mit.edu>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/namei.c b/fs/ext4/namei.c
index 0d93bc5..4db24ec 100644
--- a/fs/ext4/namei.c
+++ b/fs/ext4/namei.c
@@ -2485,8 +2485,7 @@ static int empty_dir(struct inode *inode)
    ext4_rec_len_from_disk(de1->rec_len, sb->s_blocksize);
  de = ext4_next_entry(de1, sb->s_blocksize);
  while (offset < inode->i_size) {
-  if (!bh ||
-      (void *) de >= (void *) (bh->b_data+sb->s_blocksize)) {
+  if ((void *) de >= (void *) (bh->b_data+sb->s_blocksize)) {
    unsigned int lblock;
    err = 0;
    brelse(bh);
-- 
1.7.1